DOCKER DEV ENV

amoffat opened the pull request:
Hopefully this helps others iterate quicker on their code with fewer errors.  See the [docker readme](https://github.com/amoffat/Chaos/blob/docker-dev-env/dev/docker/README.md)

It should make things as simple as going `./run.sh` and having a full chaos server, in a second, that looks at your own personal chaos fork.  This way you can test things like PRs, labels, etc, with full control on your fork, before opening a PR here.

amoffat committed:
moving dev tools into dev folder

amoffat committed:
if local secret file not found, dont use auth

amoffat committed:
disable cooldowns on no auth

amoffat committed:
derive project and owner from wherever the repo was cloned from

amoffat committed:
get urn from git itself

amoffat committed:
github user can come from urn, not the other way around

amoffat committed:
shuffling around some docker stuff

amoffat committed:
main readme for docker

amoffat committed:
updating deps

amoffat committed:
fixing paths, exposing ports

amoffat committed:
nginx config

amoffat committed:
docker readme

amoffat committed:
more readme

amoffat committed:
shuffling around readmes

amoffat committed:
undo not specifying user and pat..not needed anymore

amoffat committed:
Merge branch 'master' into docker-dev-env

amoffat committed:
Merge branch 'master' into docker-dev-env

sorcio commented on requirements.txt:
```diff
@@ -1,11 +1,17 @@
+sh==1.12.13
 six==1.10.0
-emoji==0.4.5
+Werkzeug==0.12.2
```
What are `click` and `Flask` used for?

amoffat replied:
good eye.  i was testing some stuff and :hankey: 'd my local venv requirements.  let me fix..

amoffat committed:
more accurate requirements

@sorcio caught reqs that don't need to be there, like flask and its deps.  some
of the new reqs there now are actually deps from the cryptography package that
weren't there from #227

amoffat committed:
Merge branch 'master' into docker-dev-env

xyproto commented on dev/docker/Dockerfile:
```diff
@@ -0,0 +1,70 @@
+
+RUN rm /etc/nginx/sites-enabled/default
+
+RUN printf "\
```
If `#!/bin/bash` is added here....

xyproto commented on dev/docker/Dockerfile:
```diff
@@ -0,0 +1,70 @@
+    sleep 1\n\
+
+    tail -F /var/log/supervisor/chaos-*"\
+    >> /root/start_chaos.sh
```
...and this one is piped to `/usr/bin/start_chaos` instead, followed by a `chmod +x /usr/bin/start_chaos`...

xyproto commented on dev/docker/Dockerfile:
```diff
@@ -0,0 +1,70 @@
+    >> /root/start_chaos.sh
+
+EXPOSE 80 8081
+ENTRYPOINT bash /root/start_chaos.sh
```
... then this line could just be `ENTRYPOINT start_chaos`?

amoffat replied:
Thanks @xyproto, these are good improvements.  I'll leave that for a followup PR so as to not extend voting on this 12 hour(!) PR

xyproto replied:
Understandable.

xyproto commented on dev/vagrant/README.md:
```diff
@@ -0,0 +1,13 @@
+
+    sudo su
+    cd /vagrant
+    python3 chaos.py
```
why not `sudo python3 chaos.py` instead of `sudo su`?

amoffat replied:
The vagrant README.md came from the top-level README.md without changes.  I agree we should clean it up though once this lands
